spec/models/links_udpate_spec.rb

```diff
@@ -69,12 +67,24 @@
     end
 
     it 'adds an error message' do
+      stub_request(:post, "https://publishing-api.test.gov.uk/lookup-by-base-path")
+        .with(body: "{\"base_paths\":[\"/a/base/path\"]}")
+        .to_return(status: 200, body: "{}")
+
+      tag_mapping = create(:tag_mapping, link_type: 'non-taxons')
+      links_update = build(:links_update, links: { 'non-taxons' => ['a-taxon-content-id'] })
+      links_update.tag_mappings = TagMapping.all
+
       expectation = lambda do
```

Comment:
So this expectation was here to make sure we could call this:

```
expect { expectation.call }.to change { tag_mapping.message }
```

Since you are asserting the error message on the tag mapping, both the expectation block and assertion can go away.

This means you would only need:

```
expect(tag_mapping.reload.message).to eql("We could not find this URL on GOV.UK. Invalid link types found - only taxons are allowed.")
```

We might not need to assert for equality on the exact error message, otherwise the test will fail if we change it ever so slightly. What do you think of doing something like:

```
expect(tag_mapping.reload.message).to match(/invalid link types found/i)
```